app/content/serializers/user.py

```diff
@@ -106,29 +111,52 @@ class Meta:
             "email",
             "user_class",
             "user_study",
+            "study",
+            "class_",
         )
 
     def create(self, validated_data):
-        user = User.objects.create(
-            user_id=validated_data["user_id"],
-            password=make_password(validated_data["password"]),
-            first_name=validated_data["first_name"],
-            last_name=validated_data["last_name"],
-            email=validated_data["email"],
-            user_class=validated_data["user_class"],
-            user_study=validated_data["user_study"],
-        )
-        user.set_password(validated_data["password"])
+        study = validated_data.pop("study", None)
+        class_ = validated_data.pop("class", None)
+
+        password = validated_data.pop("password")
+        user = User.objects.create(password=make_password(password), **validated_data,)
+        user.set_password(password)
+
+        if study and class_:
+            self.add_user_to_class_and_study(user, study, class_)
+
         user.save()
         return user
 
+    @staticmethod
+    def add_user_to_class_and_study(user, study, class_):
+        study = get_object_or_404(Group, slug=study)
+        class_ = get_object_or_404(Group, slug=class_)
+
+        if not study.type == GroupType.STUDY:
+            raise APIUserRegistrationException("Studie må være av typen studie.")
+
+        if not class_.type == GroupType.STUDYYEAR:
+            raise APIUserRegistrationException("Trinn må være av typen studieår.")
+
+        Membership.objects.create(user=user, group=study)
+        Membership.objects.create(user=user, group=class_)
+
     def validate_email(self, data):
         if "@ntnu.no" in data:
             raise ValidationError(
                 "Vi kan ikke sende epost til @ntnu.no-adresser, bruk @stud.ntnu.no-adressen istedenfor."
             )
         return data
 
+    def get_fields(self):
+        result = super().get_fields()
+        # Rename `class_` to `class`
+        class_ = result.pop("class_")
+        result["class"] = class_
+        return result
```

Comment:
Hvor kjøres denne?

Reply:
Det var en hacky løsning for å kunne ha feltet `class` i requesten/responsen siden `class` er et reservert keyword